src/platform/cc13xx_26xx/memory.c

```diff
@@ -0,0 +1,357 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * *  Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ * *  Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * *  Neither the name of Texas Instruments Incorporated nor the names of
+ *    its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+/*
+ *  ======== memory.c ========
+ */
+
+#if defined(__ti__) && !defined(__clang__)
+
+#pragma FUNC_EXT_CALLED(malloc);
+#pragma FUNC_EXT_CALLED(memalign);
+#pragma FUNC_EXT_CALLED(free);
+#pragma FUNC_EXT_CALLED(calloc);
+#pragma FUNC_EXT_CALLED(realloc);
+#pragma FUNC_EXT_CALLED(aligned_alloc);
+
+#define ATTRIBUTE
+
+#elif defined(__IAR_SYSTEMS_ICC__)
+
+#define ATTRIBUTE
+
+#else
+
+#define ATTRIBUTE __attribute__((used))
+
+#endif
+
+#include <errno.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <FreeRTOS.h>
+#include <portmacro.h>
+
+#if defined(__GNUC__) && !defined(__ti__)
+
+#include <reent.h>
+
+#endif
+extern void * pvPortRealloc(void * pv, size_t size);
+
+/*
+ * This Header is only needed to support advanced memory services - namely
+ * realloc() and memalign().  If the user doesn't require those features, they
+ * can remove the overhead and save both code and data.
+ */
+#if defined(TI_POSIX_FREERTOS_MEMORY_ENABLEADV)
+/*
+ * Header is a union to make sure that the size is a power of 2.
+ *
+ * On the MSP430 small model (MSP430X), size_t is 2 bytes, which makes
+ * the size of this struct 6 bytes.
+ */
+typedef union Header
+{
+    struct
+    {
+        void * actualBuf;
+        size_t size;
+    } header;
+    int pad[2]; /* 4 words on 28L, 8 bytes on most others */
+} Header;
```

Comment:
would some static_assert regarding sizes make sense and enforce limits here?

Reply:
I don't think it's necessary since it's only for the 13x4 platform and the advanced feature set isn't currently being exercised so this additional header is currently unused.